cluster-operator/src/main/java/io/strimzi/operator/cluster/model/ListenersUtils.java

```diff
@@ -0,0 +1,489 @@
+     *
+     * @param listeners List of all listeners
+     * @return          List of external listeners
+     */
+    public static List<GenericKafkaListener> externalListeners(List<GenericKafkaListener> listeners)    {
+        return listeners.stream()
+                .filter(listener -> KafkaListenerType.INTERNAL != listener.getType())
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Returns list of all Route type listeners
+     *
+     * @param listeners List of all listeners
+     * @return          List of route listeners
+     */
+    public static List<GenericKafkaListener> routeListeners(List<GenericKafkaListener> listeners)    {
+        return listeners.stream()
+                .filter(listener -> KafkaListenerType.ROUTE == listener.getType())
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Returns list of all LoadBalancer type listeners
+     *
+     * @param listeners List of all listeners
+     * @return          List of load balancer listeners
+     */
+    public static List<GenericKafkaListener> loadBalancerListeners(List<GenericKafkaListener> listeners)    {
+        return listeners.stream()
+                .filter(listener -> KafkaListenerType.LOADBALANCER == listener.getType())
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Returns list of all NodePort type listeners
+     *
+     * @param listeners List of all listeners
+     * @return          List of node port listeners
+     */
+    public static List<GenericKafkaListener> nodePortListeners(List<GenericKafkaListener> listeners)    {
+        return listeners.stream()
+                .filter(listener -> KafkaListenerType.NODEPORT == listener.getType())
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Returns list of all Ingress type listeners
+     *
+     * @param listeners List of all listeners
+     * @return          List of ingress listeners
+     */
+    public static List<GenericKafkaListener> ingressListeners(List<GenericKafkaListener> listeners)    {
+        return listeners.stream()
+                .filter(listener -> KafkaListenerType.INGRESS == listener.getType())
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Returns list of all additional DNS addresses for certificates
+     *
+     * @param listeners List of all listeners
+     * @return          List of external listeners
+     */
+    public static List<String> alternativeNames(List<GenericKafkaListener> listeners)    {
+        return listeners.stream()
+                .filter(listener -> listener.getConfiguration() != null
+                        && listener.getConfiguration().getBootstrap() != null
+                        && listener.getConfiguration().getBootstrap().getAlternativeNames() != null)
+                .map(listener -> listener.getConfiguration().getBootstrap().getAlternativeNames())
+                .flatMap(List::stream)
```

Comment:
This is just a single `flatMap(listener -> listener.getConfiguration().getBootstrap().getAlternativeNames().stream()` I think.